### PR TITLE
Support JSON Schema for REST input validation

### LINKED ISSUE #1745: JSON-schema support in autoDescribeRoute
A simple version of this would just be adding a `schema` parameter to `jsonParam` that would do runtime validation.

ping @cjh1

### PR DESCRIPTION
**Pull request by zachmullen**
Fixes #1745.

### CODE REVIEW

**Comment by zachmullen** [comment]
ping @Xarthisius @cjh1

**Comment by Xarthisius** [comment]
LGTM. Though it would be awesome if the schema was included the param's description and would show up in swagger.

**Comment by cjh1** [comment]
The problem is the schema used by swagger, is a subset of JSON schema :frowning_face:

**Comment by zachmullen** [comment]
@cjh1 do you have time to review this?

**Comment by cjh1** [comment]
@zachmullen Sure, I will take a look

File: setup.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def run(self, *arg, **kw):
     'bcrypt',
     'boto',
     'CherryPy<8',  # see https://github.com/girder/girder/issues/1615
+    'jsonschema',
```
Comment (cjh1) [inline setup.py]: Should this go in our requirements.txt?

File: girder/api/describe.py [hunk 2]
```diff
@@ -310,14 +311,18 @@ def modelParam(self, name, description=None, model=None, destName=None, paramTyp
...
-                  default=None, requireObject=False, requireArray=False):
+                  default=None, requireObject=False, requireArray=False, schema=None):
         """
         Specifies a parameter that should be processed as JSON.
 
         :param requireObject: Whether the value must be a JSON object / Python dict.
         :type requireObject: bool
         :param requireArray: Whether the value must be a JSON array / Python list.
         :type requireArray: bool
+        :param schema: A jsonschema that will be used to validate the parameter value. If
```
Comment (cjh1) [inline girder/api/describe.py]: This should probably read "A JSON Schema ..."

File: setup.py (hunk 1 above)
Comment (cjh1) [inline setup.py]: So it doesn't?

Comment (zachmullen) [inline setup.py]: Whoops, thought I had pushed my last commit onto this, turns out I did not actually commit the changes...

**Pull request merged**